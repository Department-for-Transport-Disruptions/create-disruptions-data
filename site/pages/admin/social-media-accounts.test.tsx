import renderer from "react-test-renderer";
import { describe, it, expect } from "vitest";
import SocialMediaAccounts, { SocialMediaAccountsPageProps } from "./social-media-accounts.page";

const blankInputs: SocialMediaAccountsPageProps = {
    socialMediaDetails: [],
    hootsuiteAuthUrl: "https://hootsuite-test-auth.com",
    twitterAuthUrl: "https://twitter-test-auth.com",
};

const withInputs: SocialMediaAccountsPageProps = {
    socialMediaDetails: [
        {
            id: "24858630",
            display: "testemail@gmail.com",
            accountType: "Hootsuite",
            addedBy: "Test Account",
            expiresIn: "Never",
            hootsuiteProfiles: [
                { type: "TWITTER", socialNetworkId: "1560718270506380122", id: "188295022" },
                { type: "FACEBOOK", socialNetworkId: "990838348810590", id: "138296272" },
            ],
        },
    ],
    hootsuiteAuthUrl: "https://hootsuite-test-auth.com",
    twitterAuthUrl: "https://twitter-test-auth.com",
};

describe("socialMediaAccounts", () => {
    it("should render correctly when there are no inputs", () => {
        const tree = renderer.create(<SocialMediaAccounts {...blankInputs} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should render correctly with inputs", () => {
        const tree = renderer.create(<SocialMediaAccounts {...withInputs} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
