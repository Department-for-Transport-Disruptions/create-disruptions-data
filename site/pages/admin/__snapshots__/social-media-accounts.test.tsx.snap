// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`socialMediaAccounts > should render correctly when there are no inputs 1`] = `
[
  <div
    className="app-phase-banner__wrapper"
  >
    <div
      className="govuk-phase-banner max-w-[960px] w-[90%] border-0 mx-auto app-phase-banner app-width-container"
    >
      <p
        className="govuk-phase-banner__content"
      >
        <strong
          className="govuk-tag govuk-phase-banner__content__tag"
        >
          alpha
        </strong>
        <span
          className="govuk-phase-banner__text"
        >
          This is a new service – your
           
          <a
            className="govuk-link"
            href="/feedback"
            id="feedback-link"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            feedback
          </a>
           
          will help us to improve it.
        </span>
      </p>
    </div>
  </div>,
  <div
    className="govuk-width-container"
  >
    <main
      className="govuk-main-wrapper"
      id="main-content"
    >
      <h1
        className="govuk-heading-xl"
      >
        Social media accounts
      </h1>
      <table
        className="govuk-table"
      >
        <thead
          className="govuk-table__head"
        >
          <tr
            className="govuk-table__row"
          >
            <th
              className="govuk-table__header align-middle"
              scope="col"
            >
              Account type
            </th>
            <th
              className="govuk-table__header align-middle"
              scope="col"
            >
              Username/page
            </th>
            <th
              className="govuk-table__header align-middle"
              scope="col"
            >
              Added by
            </th>
            <th
              className="govuk-table__header align-middle"
              scope="col"
            >
              Expires in
            </th>
            <th
              className="govuk-table__header align-middle"
              scope="col"
            >
              Hootsuite Profiles
            </th>
          </tr>
        </thead>
        <tbody
          className="govuk-table__body"
        />
      </table>
      <a
        className="govuk-button mt-8"
        data-module="govuk-button"
        href="https://platform.hootsuite.com/oauth2/auth?response_type=code&scope=offline&redirect_uri=http://localhost:3000/api/hootsuite-callback&client_id=6ab8fd00-4b2d-42a7-beef-8558da21c82d&state=6ab8fd00-4b2d-42a7-beef-8558da21c82d"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Connect hootsuite
      </a>
    </main>
    <div>
      <h2
        className="govuk-heading-s"
      >
        Help and Support
      </h2>
      <p
        className="govuk-body"
      >
        If you are having problems, please contact the Create Disruption Service via this link:
         
        <a
          className="govuk-link govuk-!-font-size-19"
          href="/contact"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Contact us
        </a>
      </p>
    </div>
  </div>,
  <footer
    className="govuk-footer"
    role="contentinfo"
  >
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-footer__meta"
      >
        <div
          className="govuk-footer__meta-item govuk-footer__meta-item--grow"
        >
          <h2
            className="govuk-visually-hidden"
          >
            Support links
          </h2>
          <ul
            className="govuk-footer__inline-list"
          >
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/contact"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Help
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/cookies"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Cookies
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/contact"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Contact
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/accessibility"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Accessibility
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="https://www.gov.uk/help/terms-conditions"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Terms and conditions
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/privacy"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Privacy
              </a>
            </li>
          </ul>
          <p
            className="govuk-footer__body"
          >
            Built by the
             
            <a
              className="govuk-footer__link"
              href="https://www.gov.uk/government/organisations/department-for-transport"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
              rel="noreferrer"
              target="_blank"
            >
              Department for Transport
            </a>
          </p>
          <svg
            aria-hidden="true"
            className="govuk-footer__licence-logo"
            focusable="false"
            height="17"
            viewBox="0 0 483.2 195.7"
            width="41"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
              fill="currentColor"
            />
          </svg>
          <span
            className="govuk-footer__licence-description"
          >
            All content is available under the
             
            <a
              className="govuk-footer__link"
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
              rel="license"
            >
              Open Government Licence v3.0
            </a>
            , except where otherwise stated
          </span>
        </div>
        <div
          className="govuk-footer__meta-item"
        >
          <a
            className="govuk-footer__link govuk-footer__copyright-logo bg-govuk-crest"
            href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            © Crown copyright
          </a>
        </div>
      </div>
    </div>
  </footer>,
]
`;

exports[`socialMediaAccounts > should render correctly with inputs 1`] = `
[
  <div
    className="app-phase-banner__wrapper"
  >
    <div
      className="govuk-phase-banner max-w-[960px] w-[90%] border-0 mx-auto app-phase-banner app-width-container"
    >
      <p
        className="govuk-phase-banner__content"
      >
        <strong
          className="govuk-tag govuk-phase-banner__content__tag"
        >
          alpha
        </strong>
        <span
          className="govuk-phase-banner__text"
        >
          This is a new service – your
           
          <a
            className="govuk-link"
            href="/feedback"
            id="feedback-link"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            feedback
          </a>
           
          will help us to improve it.
        </span>
      </p>
    </div>
  </div>,
  <div
    className="govuk-width-container"
  >
    <main
      className="govuk-main-wrapper"
      id="main-content"
    >
      <h1
        className="govuk-heading-xl"
      >
        Social media accounts
      </h1>
      <table
        className="govuk-table"
      >
        <thead
          className="govuk-table__head"
        >
          <tr
            className="govuk-table__row"
          >
            <th
              className="govuk-table__header align-middle"
              scope="col"
            >
              Account type
            </th>
            <th
              className="govuk-table__header align-middle"
              scope="col"
            >
              Username/page
            </th>
            <th
              className="govuk-table__header align-middle"
              scope="col"
            >
              Added by
            </th>
            <th
              className="govuk-table__header align-middle"
              scope="col"
            >
              Expires in
            </th>
            <th
              className="govuk-table__header align-middle"
              scope="col"
            >
              Hootsuite Profiles
            </th>
          </tr>
        </thead>
        <tbody
          className="govuk-table__body"
        >
          <tr
            className="govuk-table__row"
          >
            <td
              className="govuk-table__cell align-middle"
            >
              <p>
                Hootsuite
              </p>
            </td>
            <td
              className="govuk-table__cell align-middle"
            >
              <p>
                ask@iverpoolcityregion-ca-gov.uk
              </p>
            </td>
            <td
              className="govuk-table__cell align-middle"
            >
              <p>
                Chris Cavanagh
              </p>
            </td>
            <td
              className="govuk-table__cell align-middle"
            >
              <p>
                Never
              </p>
            </td>
            <td
              className="govuk-table__cell align-middle"
            >
              <li
                className="list-none"
              >
                <a
                  className="govuk-link text-govBlue"
                  href="https://twitter.com/intent/user?user_id=1234567/"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onTouchStart={[Function]}
                >
                  Twitter/43308270
                </a>
              </li>
              <li
                className="list-none"
              >
                <a
                  className="govuk-link text-govBlue"
                  href="https://twitter.com/intent/user?user_id=1234567/"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onTouchStart={[Function]}
                >
                  Twitter/29669438
                </a>
              </li>
            </td>
          </tr>
          <tr
            className="govuk-table__row"
          >
            <td
              className="govuk-table__cell align-middle"
            >
              <p>
                Hootsuite
              </p>
            </td>
            <td
              className="govuk-table__cell align-middle"
            >
              <p>
                2ask@iverpoolcityregion-ca-gov.uk
              </p>
            </td>
            <td
              className="govuk-table__cell align-middle"
            >
              <p>
                Anna Simpson
              </p>
            </td>
            <td
              className="govuk-table__cell align-middle"
            >
              <p>
                Never
              </p>
            </td>
            <td
              className="govuk-table__cell align-middle"
            >
              <li
                className="list-none"
              >
                <a
                  className="govuk-link text-govBlue"
                  href="https://twitter.com/intent/user?user_id=1234567/"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onTouchStart={[Function]}
                >
                  Twitter/43308888
                </a>
              </li>
            </td>
          </tr>
        </tbody>
      </table>
      <a
        className="govuk-button mt-8"
        data-module="govuk-button"
        href="https://platform.hootsuite.com/oauth2/auth?response_type=code&scope=offline&redirect_uri=http://localhost:3000/api/hootsuite-callback&client_id=6ab8fd00-4b2d-42a7-beef-8558da21c82d&state=6ab8fd00-4b2d-42a7-beef-8558da21c82d"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Connect hootsuite
      </a>
    </main>
    <div>
      <h2
        className="govuk-heading-s"
      >
        Help and Support
      </h2>
      <p
        className="govuk-body"
      >
        If you are having problems, please contact the Create Disruption Service via this link:
         
        <a
          className="govuk-link govuk-!-font-size-19"
          href="/contact"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Contact us
        </a>
      </p>
    </div>
  </div>,
  <footer
    className="govuk-footer"
    role="contentinfo"
  >
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-footer__meta"
      >
        <div
          className="govuk-footer__meta-item govuk-footer__meta-item--grow"
        >
          <h2
            className="govuk-visually-hidden"
          >
            Support links
          </h2>
          <ul
            className="govuk-footer__inline-list"
          >
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/contact"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Help
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/cookies"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Cookies
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/contact"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Contact
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/accessibility"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Accessibility
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="https://www.gov.uk/help/terms-conditions"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Terms and conditions
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/privacy"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Privacy
              </a>
            </li>
          </ul>
          <p
            className="govuk-footer__body"
          >
            Built by the
             
            <a
              className="govuk-footer__link"
              href="https://www.gov.uk/government/organisations/department-for-transport"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
              rel="noreferrer"
              target="_blank"
            >
              Department for Transport
            </a>
          </p>
          <svg
            aria-hidden="true"
            className="govuk-footer__licence-logo"
            focusable="false"
            height="17"
            viewBox="0 0 483.2 195.7"
            width="41"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
              fill="currentColor"
            />
          </svg>
          <span
            className="govuk-footer__licence-description"
          >
            All content is available under the
             
            <a
              className="govuk-footer__link"
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
              rel="license"
            >
              Open Government Licence v3.0
            </a>
            , except where otherwise stated
          </span>
        </div>
        <div
          className="govuk-footer__meta-item"
        >
          <a
            className="govuk-footer__link govuk-footer__copyright-logo bg-govuk-crest"
            href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            © Crown copyright
          </a>
        </div>
      </div>
    </div>
  </footer>,
]
`;
